src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -287,7 +287,7 @@ Cci.ITypeReference Cci.ITypeDefinition.GetBaseClass(EmitContext context)
             Debug.Assert(((Cci.ITypeReference)this).AsTypeDefinition(context) != null);
             NamedTypeSymbol baseType = AdaptedNamedTypeSymbol.BaseTypeNoUseSiteDiagnostics;
 
-            if (AdaptedNamedTypeSymbol.IsScriptClass)
+            if (AdaptedNamedTypeSymbol.IsScriptClass || AdaptedNamedTypeSymbol.IsExtension)
```

Comment:
>|| AdaptedNamedTypeSymbol.IsExtension [](http://example.com/codeflow?start=53&length=37)

Are we checking presence of `object` somewhere? Otherwise, let's add a PROTOTYPE comment